src/finch/interface/fuse.py

```diff
@@ -44,42 +44,55 @@
 Performance:
 ------------
 - Using `lazy` and `compute` results in faster execution due to operation fusion.
-- Different optimizers can be used with `compute`, such as the Galley optimizer, which adapts to
-  the sparsity patterns of the inputs.
-- The optimizer can be set using the `ctx` argument in `compute`, or via `set_scheduler` or `with_scheduler`.
+- Different optimizers can be used with `compute`, such as the Galley optimizer, which
+  adapts to the sparsity patterns of the inputs.
+- The optimizer can be set using the `ctx` argument in `compute`, or via `set_scheduler`
+  or `with_scheduler`.
 """
 
-def fuse(f, *args, ctx=get_default_scheduler()):
+from .lazy import compute, get_default_scheduler, lazy
+
+
+def fuse(f, *args, ctx=None):
     """
     - fuse(f, *args, ctx=get_default_scheduler()):
-        Fuses multiple array operations into a single kernel. This function allows for composing
-        operations and executing them efficiently.
+        Fuses multiple array operations into a single kernel. This function allows for
+        composing operations and executing them efficiently.
 
         Parameters:
-        - f: The function representing the operation to be fused, returning a tensor or tuple of tensor results.
+        - f: The function representing the operation to be fused, returning a tensor or
+          tuple of tensor results.
         - *args: The input arrays or LazyTensors to be fused.
-        - ctx: The scheduler to use for computation. Defaults to the result of `get_default_scheduler()`.
+        - ctx: The scheduler to use for computation. Defaults to the result of
+          `get_default_scheduler()`.
 
         Returns:
         - The result of the fused operation, a tensor or tuple of tensors.
     """
+    if ctx is None:
+        ctx = get_default_scheduler()
     args = [lazy(arg) for arg in args]
     if len(args) == 1:
         return f(args[0])
     return compute(f(*args), ctx=ctx)
 
-def fused(f, /, ctx=get_default_scheduler()):
+
+def fused(f, /, ctx=None):
```

Comment:
I think this used to be called compiled, and I'm happy to change that back if you like.